sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -214,8 +217,16 @@ private[hive] trait HiveInspectors {
 
     case c: Class[_] if c.isArray => ArrayType(javaClassToDataType(c.getComponentType))
 
+    // list type
+    case c: Class[_] if c == classOf[java.util.List[java.lang.Object]] =>
```

Comment:
The type info are erased during the compile time.
For example:
```
scala> classOf[java.util.List[_]]==classOf[java.util.List[java.lang.Object]]
res5: Boolean = true
```
Replace the `java.lang.Object` with `_`?